tests/containers/base_image_test.py

```diff
@@ -20,6 +26,84 @@
 class TestBaseImage:
     """Tests that are applicable for all images we have in this repository."""
 
+    def test_elf_files_can_link_runtime_libs(self, subtests: pytest_subtests.SubTests, image):
+        container = testcontainers.core.container.DockerContainer(image=image, user=0, group_add=[0])
+        container.with_command("/bin/sh -c 'sleep infinity'")
+
+        def check_elf_file():
+            """This python function will be executed on the image itself.
+            That's why it has to have here all imports it needs."""
+            import glob
+            import os
+            import json
+            import subprocess
+            import stat
+
+            dirs = [
+                "/bin",
+                "/lib",
+                "/lib64",
+                "/opt/app-root"
+            ]
+            for path in dirs:
+                count_scanned = 0
+                unsatisfied_deps: list[tuple[str, str]] = []
+                for dlib in glob.glob(os.path.join(path, "**"), recursive=True):
+                    # we will visit all files eventually, no need to bother with symlinks
+                    s = os.stat(dlib, follow_symlinks=False)
+                    isdirectory = stat.S_ISDIR(s.st_mode)
+                    isfile = stat.S_ISREG(s.st_mode)
+                    executable = bool(s.st_mode & (stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH))
+                    if isdirectory or not executable or not isfile:
+                        continue
+                    with open(dlib, mode='rb') as fp:
+                        magic = fp.read(4)
+                    if magic != b'\x7fELF':
```

Comment:
Similar as above - do we want to log/print files like these for the convenience?